flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/ExceptionHistoryEntryTest.java

```diff
@@ -0,0 +1,176 @@
+import org.apache.flink.runtime.taskmanager.LocalTaskManagerLocation;
+import org.apache.flink.runtime.taskmanager.TaskManagerLocation;
+import org.apache.flink.util.TestLogger;
+
+import org.junit.Test;
+
+import javax.annotation.Nullable;
+
+import java.util.Optional;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.nullValue;
+import static org.junit.Assert.assertThat;
+
+/** {@code ExceptionHistoryEntryTest} tests the instantiation of {@link ExceptionHistoryEntry}. */
+public class ExceptionHistoryEntryTest extends TestLogger {
+
+    @Test
+    public void testFromGlobalFailure() {
+        final Throwable failureCause = new RuntimeException("failure cause");
+        final long timestamp = System.currentTimeMillis();
+
+        final ExceptionHistoryEntry testInstance =
+                ExceptionHistoryEntry.fromGlobalFailure(failureCause, timestamp);
+
+        assertThat(
+                testInstance.getException().deserializeError(ClassLoader.getSystemClassLoader()),
+                is(failureCause));
+        assertThat(testInstance.getTimestamp(), is(timestamp));
+        assertThat(testInstance.getFailingTaskName(), is(nullValue()));
+        assertThat(testInstance.getAssignedResourceLocation(), is(nullValue()));
+    }
+
+    @Test
+    public void testFromFailedExecution() {
+        final Throwable failureCause = new RuntimeException("Expected failure");
+        final long failureTimestamp = System.currentTimeMillis();
+        final String taskNameWithSubTaskIndex = "task name";
+        final TaskManagerLocation taskManagerLocation = new LocalTaskManagerLocation();
+
+        final ExceptionHistoryEntry testInstance =
+                ExceptionHistoryEntry.fromFailedExecution(
+                        createExecutionWithFailure(
+                                failureCause, failureTimestamp, taskManagerLocation),
+                        taskNameWithSubTaskIndex);
+
+        assertThat(
+                testInstance.getException().deserializeError(ClassLoader.getSystemClassLoader()),
+                is(failureCause));
+        assertThat(testInstance.getTimestamp(), is(failureTimestamp));
+        assertThat(testInstance.getFailingTaskName(), is(taskNameWithSubTaskIndex));
+        assertThat(testInstance.getAssignedResourceLocation(), is(taskManagerLocation));
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testFromFailedExecutionWithoutFailure() {
+        ExceptionHistoryEntry.fromFailedExecution(createExecutionWithoutFailure(), "task name");
+    }
+
+    /**
+     * Creates a {@link TestingExecution} that overwrites the relevant methods to emulate a {@link
+     * AccessExecution} without a failure.
+     *
+     * @return The {@code TestingExecution} instance.
+     */
+    private static TestingExecution createExecutionWithoutFailure() {
+        return new TestingExecution(null, new LocalTaskManagerLocation());
```

Comment:
Fair enough. Firstly, I wanted to oppose this because I think that the method name helps understanding what the constructor call does. But using a proper local variable name is good enough, I guess.